eokwukwe/ecommerce-shop-api
# #169659876 User Login with Facebook

#### What does this PR do?
Implement user login with Facebook
 #### Description of Task to be completed?
- [x] Implement the controller functionality for a user to login with Facebook
 #### How should this be manually tested?
- clone the branch
- Run `npm install` to install dependencies
- Run `npm run dev` to start the development server
- Go to `https://developers.facebook.com/tools/explorer` and generate a facebook `accessToken`
- Send a `POST` request to `http://localhost:80/api/v1/customers/facebook` with the following credentials:
```
{
  "access_token": "<facebook accessToken>"
}
```
- Expected output if the credentials are correct: 
```
{
  "customer": {
    "schema": {
      "customer_id": <id>,
      "name": "<name>",
      "email": "<email>",
      "address_1": null,
      "address_2": null,
      "city": null,
      "region": null,
      "postal_code": null,
      "country": null,
      "shipping_region_id": 1,
      "day_phone": null,
      "eve_phone": null,
      "mob_phone": null,
      "credit_card": null
    }
  },
  "accessToken": "Bearer <token.....>",
  "expires_in": "24h"
}
```
 #### Any background context you want to provide?
N/A
 #### What are the relevant pivotal tracker stories?
[#169659876 ](https://www.pivotaltracker.com/story/show/169659876)
